resources/guides/first_year_guide.md

```diff
@@ -0,0 +1,221 @@
+    -   `cd ~` will take you to your Home folder.
+    -   `cd ..` will take you back out of the folder you are currently in.
+-   `echo "Chip the Cool Cat"` will display "Chip the Cool Cat" in your terminal.
+-   `cat fileName` will display the contents of a the file named "fileName".
+
+Some tips:
+
+-   You can scroll through your command history using the up and down arrows.
+-   **ctrl+c** can be used to kill a process that is running in your shell.
+
+### C Compiler: GCC
+
+#### Description
+
+When you write C code, you need a way to make it **executable** on your system. This is where a compiler comes in: it reads in the more human-friendly C code and outputs a program that your computer can read and run.
+
+GCC is the compiler you should use when developing C code for your classes, unless your professor specifies otherwise. When you compile code using a different compiler, or on a different system, your program may behave differently than it will on the school's system using GCC. For this reason, it is also sometimes recommended that you double check your programs on the School of Computer Science's systems before submitting more complicated assignments.
+
+#### Installing
+
+On Windows running WSL with Ubuntu (the default), enter the following into your shell:
+
+```
+sudo apt-get install gcc
+```
+
+On Mac, [this video](https://youtu.be/0z-fCNNqfEg) provides one way of installing GCC, as well as the Homebrew package manager.
+
+#### Getting Started
+
+This section will help you create a sample C program, compile it, and run it.
+Copy and paste the following code into a new file on your system. Call it "hello.c" and remember where you save it.
+
+```c
+#include <stdio.h>
+
+int main(void) {
+    printf("Chip says hello!");
+}
+```
+
+Find the file in your shell. For example, if you put the file on your Desktop, use the following command to go to there.
+
+```sh
+cd ~/Desktop
+```
+
+Compile the program into a **binary** (runnable file, executable, etc.).
+
+-   `-Wall` tells the compiler to output all warnings.
+-   `-o hello.out` tells the compiler to save the created binary with the name "hello.out".
+
+```sh
+gcc -Wall hello.c -o hello.out
+```
+
+Run the binary you just created.
+
+```sh
+./hello.out
+```
+
+The output should be
+
+```
+Chip says hello!
+```
+
+### Remote Connection: SSH and NoMachine
+
+#### Description
+
+As a computer science student, you have access to the School of Computer Science's servers, which provide software in a Linux environment that you are free to use. On campus, they can be accessed using School of Computer Science computer labs. Remotely, they can be accessed in several ways.
+
+-   SSH (recommended): Remotely use a shell on the server.
+-   NoMachine: Remote desktop environment.
+
+:::important
+You must [authenticate your UWin account](https://auth.cs.uwindsor.ca/) with the School of Computer Science before you can log in remotely.
+:::
+
+#### SSH
+
```

Comment:
not an issue but just something to keep in mind. currently students need to connect to the vpn to be able to ssh into the cs servers